src/components/routerLink.ts

```diff
@@ -0,0 +1,36 @@
+import { PropType, defineComponent, h } from 'vue'
+import { useRouter } from '@/compositions/useRouter'
+import { RouteMethod } from '@/types/routeMethod'
+
+export default defineComponent({
+  name: 'RouterLink',
+  expose: [],
+  props: {
+    to: {
+      type: Function as PropType<() => ReturnType<RouteMethod>>,
+      required: true,
```

Comment:
I'm assuming one of the things this doesn't account for yet are the other types we will accept for `to`, like a string?

Reply:
Correct, we'll have to make to a union.